### PR TITLE
Feature/UI testing

### PR DESCRIPTION
**Pull request by rsommerfeld**
Add capability to test UI features. The pipeline will now download folders from telegramTestResults.

Also, there is a new class TCTMocks that unifies all different kinds of default mock values and methods to create mock objects.

### CODE REVIEW

File: packages/TelegramClient-Tests.package/TCTAcceptanceTests.class/instance/testMessageAuthors.st [hunk 1]
```diff
@@ -4,14 +4,14 @@ testMessageAuthors
 	| message user chat_id notification |
 	self receivedMessage: nil.
 	chat_id := ((self core chats) first) id.
-	message := self class mockMessageFrom: chat_id.
+	message := TCTMocks mockMessageEventFrom: chat_id.
 
 	self core handleEvent: message.
-	user := self class mockUserWith: (self class mockMessageID asString, chat_id).
+	user := TCTMocks mockUserWith: (TCTMocks mockMessageID asString, chat_id).
```
Comment (permler) [inline packages/TelegramClient-Tests.package/TCTAcceptanceTests.class/instance/testMessageAuthors.st:10]: Why do we mock a User with a MessageID?

File: packages/TelegramClient-Tests.package/TCTUITest.class/instance/assert.looksLike..st [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+assert: aMorph looksLike: aScreenshotFilename
+	
+	| screenshotHash morphHash |
+	
+	screenshotHash := self computeHashOf: (ImageReadWriter formFromFileNamed: aScreenshotFilename).
+	morphHash := self computeHashOf: (self takeScreenshotOf: aMorph).
+	
+	"only comparing hashes might lead to false positives"
+	screenshotHash = morphHash ifFalse: 
+			[AssertionFailure signal: 'Assertion failed']
```
Comment (permler) [inline packages/TelegramClient-Tests.package/TCTUITest.class/instance/assert.looksLike..st:11]: Maybe we can use an AssertionFailure that give the developer more useful information.

File: packages/TelegramClient-Tests.package/TCTUITest.class/instance/takeScreenshotOf..st [hunk 3]
```diff
@@ -0,0 +1,4 @@
+image handling
+takeScreenshotOf: aMorph
+	
+	^ aMorph imageForm: 32 backgroundColor: Color black forRectangle: aMorph bounds
```
Comment (permler) [inline packages/TelegramClient-Tests.package/TCTUITest.class/instance/takeScreenshotOf..st:4]: Are that magic numbers?

File: packages/TelegramClient-Tests.package/TCTUITest.class/instance/assert.looksLike..st (hunk 2 above)
Comment (rsommerfeld) [reply packages/TelegramClient-Tests.package/TCTUITest.class/instance/assert.looksLike..st:11]: Done

File: packages/TelegramClient-Tests.package/TCTUITest.class/instance/takeScreenshotOf..st (hunk 3 above)
Comment (rsommerfeld) [reply packages/TelegramClient-Tests.package/TCTUITest.class/instance/takeScreenshotOf..st:4]: Converted to default values

File: packages/TelegramClient-Tests.package/TCTAcceptanceTests.class/instance/testMessageAuthors.st (hunk 1 above)
Comment (rsommerfeld) [reply packages/TelegramClient-Tests.package/TCTAcceptanceTests.class/instance/testMessageAuthors.st:10]: That has nothing to do with the new code. What is happening here, is that you have to give the extra attribute and that has to match what was already supplied in the initial new message event. However, those are specifics of that test (which was written by you 😛 )

File: packages/TelegramClient-Tests.package/TCTUITest.class/instance/forceSaveScreenshotOf.as.in..st [hunk 4]
```diff
@@ -0,0 +1,10 @@
+image handling
+forceSaveScreenshotOf: aMorph as: aString in: anotherString
```
Comment (tom-richter) [inline packages/TelegramClient-Tests.package/TCTUITest.class/instance/forceSaveScreenshotOf.as.in..st:2]: We could rename aString to fileName and anotherString to folderName

**Review by tom-richter — CHANGES_REQUESTED** [review]

File: packages/TelegramClient-Tests.package/TCTUITest.class/instance/forceSaveScreenshotOf.as.in..st (hunk 4 above)
Comment (rsommerfeld) [reply packages/TelegramClient-Tests.package/TCTUITest.class/instance/forceSaveScreenshotOf.as.in..st]: Done

**Review by tom-richter — APPROVED** [review]

**Pull request merged**